include/openmc/nuclide.h

```diff
@@ -387,11 +397,9 @@ class Nuclide {
         // This guarantees the randomness and, at the same time, makes sure we
         // reuse random numbers for the same nuclide at different temperatures,
         // therefore preserving correlation of temperature in probability tables.
-        p.stream_ = STREAM_URR_PTABLE;
         //TODO: to maintain the same random number stream as the Fortran code this
         //replaces, the seed is set with index_ + 1 instead of index_
-        double r = future_prn(static_cast<int64_t>(index_ + 1), *p.current_seed());
-        p.stream_ = STREAM_TRACKING;
+        double r = future_prn(static_cast<int64_t>(index_ + 1), p.seeds_[STREAM_URR_PTABLE]);
```

Comment:
Ought to consider this change on the main branch :+1:

Reply:
For sure!